src/redux/reducer.js

```diff
@@ -59,15 +61,47 @@ function loadApi(state = initialState.get('loadApi'), action) {
 
 function REPL(state = initialState.get('REPL'), action) {
   switch (action.type) {
+    case actType.GET_PREV_REPL_CODE:
+      //if codeIndex hasn't reached the begining of history and
+      //the previous code isn't an empty string
+      if (state.get('codeIndex') < state.get('history').size &&
```

Comment:
The idea was that the code would reflect how the REPL is set up visually. So if the repl is a list, and the codeIndex is 0, then the code will reflect what is on the repl input line. If it goes up to 1, then it'll show what is on the line before it, and so on and so on. It isn't as clean as that in the code since the input code isn't actually a part of the repl history list, and accessing the 0th element of the list actually means accessing the last element in the repl history. If each entry was inserted at the beginning, rather than pushed onto the end that would make it more reflective. 

But my overall reasoning was that codeIndex refers to the visual list that is created when you use the REPL, rather than strictly the list history. That's why it's code index instead of say historyIndex. Does that clear it up?